packages/shaderlab/src/shaders/pbr.gs

```diff
@@ -1,103 +1,85 @@
+        material_ClearCoatRoughness("ClearCoatRoughness", Range(0, 1, 0.01)) = 0;
         material_ClearCoatRoughnessTexture("ClearCoatRoughnessTexture", Texture2D);
         material_ClearCoatNormalTexture("ClearCoatNormalTexture", Texture2D);
       }
+
+      Header("Common") {
+        material_AlphaCutoff( "AlphaCutoff", Range(0, 1, 0.01) ) = 0;
+        material_TilingOffset("TilingOffset", Vector4) = (1, 1, 0, 0);
+      }
     }
 
     EditorMacros {
       Header("Conditional Macors") {
-        MATERIAL_OMIT_NORMAL("OMIT_NORMAL");
-        MATERIAL_ENABLE_CLEAR_COAT("ENABLE_CLEAR_COAT");
         MATERIAL_HAS_BASETEXTURE("HAS_BASETEXTURE");
+        MATERIAL_HAS_ROUGHNESS_METALLIC_TEXTURE("HAS_ROUGHNESS_METALLIC_TEXTURE");
+        MATERIAL_ENABLE_ANISOTROPY("ENABLE_ANISOTROPY");
+        MATERIAL_HAS_ANISOTROPY_TEXTURE("HAS_ANISOTROPY_TEXTURE")
         MATERIAL_HAS_NORMALTEXTURE("HAS_NORMALTEXTURE");
         MATERIAL_HAS_EMISSIVETEXTURE("HAS_EMISSIVETEXTURE");
-        MATERIAL_HAS_ROUGHNESS_METALLIC_TEXTURE("HAS_ROUGHNESS_METALLIC_TEXTURE");
         MATERIAL_HAS_OCCLUSION_TEXTURE("HAS_OCCLUSION_TEXTURE");
+        MATERIAL_ENABLE_CLEAR_COAT("ENABLE_CLEAR_COAT");
         MATERIAL_HAS_CLEAR_COAT_TEXTURE("HAS_CLEAR_COAT_TEXTURE");
         MATERIAL_HAS_CLEAR_COAT_ROUGHNESS_TEXTURE("HAS_CLEAR_COAT_ROUGHNESS_TEXTURE");
         MATERIAL_HAS_CLEAR_COAT_NORMAL_TEXTURE("HAS_CLEAR_COAT_NORMAL_TEXTURE");
-        MATERIAL_IS_ALPHA_CUTOFF("IS_ALPHA_CUTOFF");
         MATERIAL_IS_TRANSPARENT("IS_TRANSPARENT");
+        MATERIAL_IS_ALPHA_CUTOFF("IS_ALPHA_CUTOFF");
       }
 
-      Collapsible("Enabled Macros") {
+      Header("Enabled Macros") {
         [On] MATERIAL_NEED_WORLD_POS("WORLD POS");
         [On] MATERIAL_NEED_TILING_OFFSET("TILING_OFFSET");
       }
     }
     
     SubShader "Default" {
-      Pass "Pass0" {
+      UsePass "pbr/Default/ShadowCaster"
+
+      Pass "Forward Pass" {
         Tags { pipelineStage = "Forward"} 
 
         #define IS_METALLIC_WORKFLOW
+        #include "forwardPassPBR.glsl"
 
-        struct _galacean_a2v {
-          #include "attrib.glsl"
-        }
-
-        struct _galacean_v2f {
-          #include "varying.glsl"
-        }
-
-        #include "globals.glsl"
-
-        VertexShader = pbrVert;
-        FragmentShader = pbrFrag;
-
-        _galacean_v2f pbrVert(_galacean_a2v attr) {
-          _galacean_v2f v;
-
-          #include "pbr_vert1.glsl"
-
-          #include "blendShape_vert.glsl"
-          #include "skinning_vert.glsl"
-
-          #include "pbr_vert2.glsl"
-
-          #include "ShadowVertex.glsl"
-          #include "FogVertex.glsl"
-
-          return v;
-        }
-
-        void pbrFrag(_galacean_v2f v) {
-          #include "pbr_frag.glsl"
-          #include "FogFragment.glsl"
-
-          #ifndef ENGINE_IS_COLORSPACE_GAMMA
-              gl_FragColor = linearToGamma(gl_FragColor);
-          #endif
-        }
       }
```

Comment:
why put `VertexShader = pbrVert;
FragmentShader = pbrFrag;` in `forwardPassPBR.glsl`？
```suggestion
      }
```

Reply:
done